packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx

```diff
@@ -1,56 +1,73 @@
 import React from 'react';
 import { Portal } from '../../Portal/Portal';
-import { usePlotContext, usePlotData } from '../context';
+import { usePlotContext } from '../context';
 import { CursorPlugin } from './CursorPlugin';
 import { SeriesTable, SeriesTableRowProps } from '../../Graph/GraphTooltip/SeriesTable';
-import { FieldType, formattedValueToString, getDisplayProcessor, getFieldDisplayName, TimeZone } from '@grafana/data';
+import {
+  DataFrame,
+  FieldType,
+  formattedValueToString,
+  getDisplayProcessor,
+  getFieldDisplayName,
+  TimeZone,
+} from '@grafana/data';
 import { TooltipContainer } from '../../Chart/TooltipContainer';
 import { TooltipMode } from '../../Chart/Tooltip';
+import { useGraphNGContext } from '../../GraphNG/hooks';
 
 interface TooltipPluginProps {
   mode?: TooltipMode;
   timeZone: TimeZone;
+  data: DataFrame[];
 }
 
 /**
  * @alpha
  */
-export const TooltipPlugin: React.FC<TooltipPluginProps> = ({ mode = 'single', timeZone }) => {
+export const TooltipPlugin: React.FC<TooltipPluginProps> = ({ mode = 'single', timeZone, ...otherProps }) => {
   const pluginId = 'PlotTooltip';
   const plotContext = usePlotContext();
-  const { data, getField, getXAxisFields } = usePlotData();
+  const graphContext = useGraphNGContext();
 
-  const xAxisFields = getXAxisFields();
-  // assuming single x-axis
-  const xAxisField = xAxisFields[0];
-  const xAxisFmt = xAxisField.display || getDisplayProcessor({ field: xAxisField, timeZone });
+  let xField = graphContext.getXAxisField(otherProps.data);
```

Comment:
@ryantxu using the matchers/dim fields approach you added in XYChart. GraphNG context exposes the corresponding matchers as well as method for uPlot's series index -> dataframe/field index maping